src/core/src/bootstrap/ConfigurationFactory.py

```diff
@@ -240,4 +254,43 @@ def new_test_configuration(self, package_manager_name, package_manager_component
         configuration['config_env'] = Constants.TEST
         # perform desired modifications to configuration
         return configuration
+
+    def get_lifecycle_manager_component(self, vm_cloud_type):
+        """ finding life cycle manager based on vm and returning component name added in the prod configuration """
+        azure_lifecycle_manager_component = LifecycleManagerAzure
+        arc_lifecycle_manager_component = LifecycleManagerArc
+        if (vm_cloud_type == Constants.VMCloudType.AZURE):
+            return azure_lifecycle_manager_component
+        elif (vm_cloud_type == Constants.VMCloudType.ARC):
+            return arc_lifecycle_manager_component
+        
+        return azure_lifecycle_manager_component
+   
+    def get_vm_cloud_type(self):
+        """ detects vm type. logic taken from HCRP code: https://github.com/PowerShell/DesiredStateConfiguration/blob/dev/src/dsc/dsc_service/service_main.cpp#L115
+        Todo:  how to check this only when it is Auto Assessment operation??? """
+        metadata_value = "True"
+        user_agent_value = "ArcAgent"
+        request = urlreq.Request(Constants.IMDS_END_POINT)
+        request.add_header('Metadata', metadata_value)
+        request.add_header('UserAgent', user_agent_value)
+        print("\nTrying to connect IMDS end point. URL:{0}.".format(str(Constants.IMDS_END_POINT)))
+        for i in range(0, Constants.IMDS_CONNECTION_RETRY_COUNT + 1):
```

Comment:
Rename to MAX_IMDS_CONNECTION_RETRY_COUNT since it's the maximum times we will try connecting. IMDS_CONNECTION_RETRY_COUNT might be interpreted as the current retry count or iteration#